core/src/main/scala/cats/Show.scala

```diff
@@ -27,7 +32,7 @@ object Show {
 
   final case class Shown(override val toString: String) extends AnyVal
   object Shown {
-    implicit def mat[A](x: A)(implicit z: Show[A]): Shown = Shown(z show x)
+    implicit def mat[A](x: A)(implicit z: ContravariantShow[A]): Shown = Shown(z show x)
```

Comment:
I wonder if we can get the test to pass by asking for `Show[_ >: A]` which is effectively what this trick is doing, and not having to introduce the new type. I *think* we can (at least on 2.12).